src/models/CommonModel.ts

```diff
@@ -25,4 +25,76 @@ export class CommonModel extends CommonSchema<CommonModel>{
         }
         return newCommonModel;
     }
+
+
+    /**
+     * Only merge if left side is undefined and right side is sat OR both sides are defined
+     * 
+     * @param mergeTo CommonModel to merge into
+     * @param mergeFrom CommonModel to merge values from
+     */
+    static mergeCommonModels(mergeTo: CommonModel, mergeFrom: CommonModel, originalSchema: Schema) : CommonModel {
+        const mergeToProperties = mergeTo.properties;
+        const mergeFromProperties = mergeFrom.properties;
+        if(mergeFromProperties !== undefined){
+            if(mergeToProperties === undefined){
+                mergeTo.properties = mergeFromProperties;
+            } else if(mergeToProperties !== undefined && mergeFromProperties !== undefined) {
+                Object.entries(mergeFromProperties).forEach(([propName, prop])=> {
+                    if(mergeToProperties![propName] !== undefined){
+                      mergeTo.properties![propName] = CommonModel.mergeCommonModels(mergeToProperties![propName], prop, originalSchema);
+                    } else {
+                      mergeTo.properties![propName] = prop;
+                    }
+                });
+            }
+        }
+
+        if(mergeTo.items === undefined && mergeFrom.items !== undefined){
```

Comment:
I forgot to mention that if you apply changes, you can add them here too, I mean changed if-s with undefined

Reply:
ahh! right, 2 sec